src/api/app/controllers/webui/project_controller.rb

```diff
@@ -786,15 +791,8 @@ def monitor_set_filter(defaults)
     @status_filter = []
     @avail_status_values.each do |s|
       id = s.delete(' ')
-      if params.key?(id)
-        next unless (begin
-                       Integer(params[id])
-                     rescue ArgumentError
-                       1
-                     end) > 0
-      else
-        next unless defaults
-      end
+      next unless params[id].try_to_i(1) > 0
```

Comment:
Why does it have to return `1`? Can't we change the condition? We are not using the returned value anyway.

Reply:
`params.compact.fetch(id, 1).to_i` to get rid of the `nil` values.

Reply:
I really want to avoid to have the code for `try_to_i` because it's possible to do without it.

Reply:
It would work like this although I find it super ugly:
```
id = s.delete(' ')
      if params.key?(id)
        next unless (begin
                       Integer(params.compact.fetch(id, 1))
                     rescue ArgumentError
                       1
                     end) > 0
```

Reply:
How about `(params[id] || 1).to_i`?

Loading all these modules for just one occasion feels a bit like "Mit Kanonen auf Spatzen schiessen"

Reply:
> (params[id] || 1).to_i

return also 0 for strings. We have two cases to cover: nil and a string which is not a number!